packages/template-retail-react-app/app/components/display-price/index.jsx

```diff
@@ -22,34 +22,41 @@ const DisplayPrice = ({
 }) => {
     const intl = useIntl()
     const {currency: activeCurrency} = useCurrency()
+    const discountPriceText = intl.formatNumber(discountPrice, {
+        style: 'currency',
+        currency: currency || activeCurrency
+    })
+    const basePriceText = intl.formatNumber(basePrice, {
+        style: 'currency',
+        currency: currency || activeCurrency
+    })
+
     return (
         <Skeleton isLoaded={basePrice} display={'flex'} {...skeletonProps}>
-            <Text fontWeight="bold" fontSize="md" mr={1}>
-                {isProductASet &&
-                    `${intl.formatMessage({
+            {isProductASet && (
+                <Text fontWeight="bold" fontSize="md" mr={1}>
+                    {intl.formatMessage({
                         id: 'product_view.label.starting_at_price',
                         defaultMessage: 'Starting at'
-                    })} `}
-            </Text>
-            {typeof discountPrice === 'number' && (
-                <Text as="b" {...discountPriceProps}>
-                    {intl.formatNumber(discountPrice, {
-                        style: 'currency',
-                        currency: currency || activeCurrency
                     })}
                 </Text>
             )}
-            <Text
-                as={typeof discountPrice === 'number' ? 's' : 'b'}
-                ml={typeof discountPrice === 'number' ? 2 : 0}
-                fontWeight={discountPrice ? 'normal' : 'bold'}
-                {...basePriceProps}
-            >
-                {intl.formatNumber(basePrice, {
-                    style: 'currency',
-                    currency: currency || activeCurrency
-                })}
-            </Text>
+            {typeof discountPrice === 'number' && (
+                <Text as="b" {...discountPriceProps} aria-label={`Sale price ${discountPriceText}`}>
+                    {discountPriceText}
+                </Text>
+            )}
+            {basePrice > discountPrice && (
+                <Text
+                    aria-label={`Original price ${basePriceText}`}
+                    as={typeof discountPrice === 'number' ? 's' : 'b'}
```

Comment:
In what case would `discountPrice` be a non number? Asking because with the current implemented logic, if `discountPrice` is not a number and if `basePrice > discountPrice` evaluates to false, this component doesn't render any price

Reply:
When discountPrice is null and basePrice is defined, it will still render basePrice

Reply:
I think @joeluong-sfcc is on to something here. The way we have this set up doesn't make it very readable which leads to having to ask questions to the developer about that is going on instead of understanding it first hand. 

One suggestion I could make is if we used intermediary variables to define something like "currentPrice" and "oldPrice" (these names aren't necessarily the best) then we could have JSX that would always render the current price, and conditionally render the old price. Making it easier to understand what we are accomplishing.

Reply:
yes we had that exact chat yesterday, but we was wondering if that would be a breaking change. Will change this when we get an agreement around breaking change or not.